Use more smart pointers in editing code

cdumez opened the pull request:
#### 46e64865e4eda18aa70e6e8061ffd78359f57b63
<pre>
Use more smart pointers in editing code
<a href="https://bugs.webkit.org/show_bug.cgi?id=261844">https://bugs.webkit.org/show_bug.cgi?id=261844</a>

Reviewed by Brent Fulgham.

* Source/WebCore/editing/ApplyBlockElementCommand.cpp:
(WebCore::ApplyBlockElementCommand::ApplyBlockElementCommand):
* Source/WebCore/editing/ApplyBlockElementCommand.h:
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::ApplyStyleCommand):
* Source/WebCore/editing/ApplyStyleCommand.h:
(WebCore::ApplyStyleCommand::create):
* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::BreakBlockquoteCommand):
* Source/WebCore/editing/BreakBlockquoteCommand.h:
(WebCore::BreakBlockquoteCommand::create):
* Source/WebCore/editing/ChangeListTypeCommand.h:
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::CompositeEditCommand):
(WebCore::CompositeEditCommand::apply):
(WebCore::CompositeEditCommand::ensureComposition):
(WebCore::CompositeEditCommand::applyStyle):
(WebCore::CompositeEditCommand::insertParagraphSeparator):
(WebCore::CompositeEditCommand::insertLineBreak):
(WebCore::CompositeEditCommand::inputText):
(WebCore::CompositeEditCommand::deleteSelection):
(WebCore::CompositeEditCommand::deleteInsignificantText):
(WebCore::CompositeEditCommand::appendBlockPlaceholder):
(WebCore::CompositeEditCommand::insertBlockPlaceholder):
(WebCore::CompositeEditCommand::addBlockPlaceholderIfNeeded):
(WebCore::CompositeEditCommand::insertNewDefaultParagraphElementAt):
(WebCore::CompositeEditCommand::moveParagraphContentsToNewBlockIfNecessary):
(WebCore::CompositeEditCommand::moveParagraphWithClones):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/CompositeEditCommand.h:
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::CreateLinkCommand):
* Source/WebCore/editing/CreateLinkCommand.h:
(WebCore::CreateLinkCommand::create):
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::DeleteSelectionCommand::DeleteSelectionCommand):
(WebCore::DeleteSelectionCommand::removeNode):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
* Source/WebCore/editing/DeleteSelectionCommand.h:
(WebCore::DeleteSelectionCommand::create):
* Source/WebCore/editing/DictationCommand.cpp:
(WebCore::DictationCommand::DictationCommand):
(WebCore::DictationCommand::insertText):
(WebCore::DictationCommand::insertParagraphSeparator):
* Source/WebCore/editing/DictationCommand.h:
(WebCore::DictationCommand::create):
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::EditCommand::EditCommand):
(WebCore::SimpleEditCommand::SimpleEditCommand):
* Source/WebCore/editing/EditCommand.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::ClearTextCommand::ClearTextCommand):
(WebCore::ClearTextCommand::CreateAndApply):
(WebCore::TemporarySelectionChange::TemporarySelectionChange):
(WebCore::Editor::selectionForCommand):
(WebCore::Editor::behavior const):
(WebCore::Editor::client const):
(WebCore::Editor::canEdit const):
(WebCore::Editor::canEditRichly const):
(WebCore::Editor::canDHTMLCut):
(WebCore::Editor::canDHTMLCopy):
(WebCore::Editor::canCopy const):
(WebCore::Editor::canPaste const):
(WebCore::Editor::canDelete const):
(WebCore::Editor::shouldSmartDelete):
(WebCore::Editor::deleteWithDirection):
(WebCore::Editor::deleteSelectionWithSmartDelete):
(WebCore::Editor::clearText):
(WebCore::Editor::pasteAsPlainText):
(WebCore::Editor::pasteAsPlainTextBypassingDHTML):
(WebCore::Editor::replaceSelectionWithFragment):
(WebCore::Editor::selectedRange):
(WebCore::Editor::tryDHTMLCopy):
(WebCore::Editor::tryDHTMLCut):
(WebCore::Editor::shouldInsertText const):
(WebCore::Editor::hasBidiSelection const):
(WebCore::Editor::selectionUnorderedListState const):
(WebCore::Editor::selectionOrderedListState const):
(WebCore::Editor::insertOrderedList):
(WebCore::Editor::insertUnorderedList):
(WebCore::Editor::increaseSelectionListLevel):
(WebCore::Editor::increaseSelectionListLevelOrdered):
(WebCore::Editor::increaseSelectionListLevelUnordered):
(WebCore::Editor::decreaseSelectionListLevel):
(WebCore::Editor::removeFormattingAndStyle):
(WebCore::Editor::findEventTargetFromSelection const):
(WebCore::Editor::applyStyle):
(WebCore::Editor::applyParagraphStyle):
(WebCore::Editor::applyStyleToSelection):
(WebCore::Editor::applyParagraphStyleToSelection):
(WebCore::Editor::selectionStartHasStyle const):
(WebCore::Editor::selectionHasStyle const):
(WebCore::Editor::selectionStartCSSPropertyValue):
(WebCore::Editor::indent):
(WebCore::Editor::outdent):
(WebCore::Editor::appliedEditing):
(WebCore::Editor::unappliedEditing):
(WebCore::Editor::reappliedEditing):
(WebCore::Editor::clear):
(WebCore::Editor::insertText):
(WebCore::Editor::insertTextForConfirmedComposition):
(WebCore::Editor::insertTextWithoutSendingTextEvent):
(WebCore::Editor::insertLineBreak):
(WebCore::Editor::insertParagraphSeparator):
(WebCore::Editor::insertParagraphSeparatorInQuotedContent):
(WebCore::Editor::canCopyFont const):
(WebCore::Editor::performCutOrCopy):
(WebCore::Editor::paste):
(WebCore::Editor::pasteAsQuotation):
(WebCore::Editor::quoteFragmentForPasting):
(WebCore::Editor::changeSelectionListType):
(WebCore::Editor::simplifyMarkup):
(WebCore::Editor::copyURL):
(WebCore::Editor::copyImage):
(WebCore::Editor::revealSelectionIfNeededAfterLoadingImageForElement):
(WebCore::Editor::renderLayerDidScroll):
(WebCore::Editor::registerCustomUndoStep):
(WebCore::Editor::setBaseWritingDirection):
(WebCore::Editor::baseWritingDirectionForSelectionStart const):
(WebCore::Editor::selectComposition):
(WebCore::Editor::confirmOrCancelCompositionAndNotifyClient):
(WebCore::SetCompositionScope::SetCompositionScope):
(WebCore::Editor::setComposition):
(WebCore::Editor::ignoreSpelling):
(WebCore::Editor::learnSpelling):
(WebCore::Editor::advanceToNextMisspelling):
(WebCore::Editor::misspelledWordAtCaretOrRange const):
(WebCore::Editor::guessesForMisspelledWord const):
(WebCore::Editor::guessesForMisspelledOrUngrammatical):
(WebCore::Editor::markMisspellingsAfterTypingToWord):
(WebCore::Editor::isSpellCheckingEnabledInFocusedNode const):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
(WebCore::Editor::markAndReplaceFor):
(WebCore::Editor::updateMarkersForWordsAffectedByEditing):
(WebCore::Editor::rangeForPoint):
(WebCore::Editor::revealSelectionAfterEditingOperation):
(WebCore::Editor::setIgnoreSelectionChanges):
(WebCore::Editor::getCompositionSelection const):
(WebCore::Editor::transpose):
(WebCore::Editor::changeSelectionAfterCommand):
(WebCore::Editor::selectedText const):
(WebCore::Editor::selectedTextForDataTransfer const):
(WebCore::Editor::insertTextPlaceholder):
(WebCore::Editor::removeTextPlaceholder):
(WebCore::Editor::shouldChangeSelection const):
(WebCore::Editor::computeAndSetTypingStyle):
(WebCore::Editor::findString):
(WebCore::Editor::rangeOfString):
(WebCore::Editor::countMatchesForText):
(WebCore::Editor::respondToChangedSelection):
(WebCore::Editor::shouldDetectTelephoneNumbers const):
(WebCore::Editor::scanSelectionForTelephoneNumbers):
(WebCore::Editor::editorUIUpdateTimerFired):
(WebCore::Editor::selectionStartHasMarkerFor const):
(WebCore::Editor::stringForCandidateRequest const):
(WebCore::Editor::contextRangeForCandidateRequest const):
(WebCore::Editor::fontAttributesAtSelectionStart):
(WebCore::Editor::notifyClientOfAttachmentUpdates):
(WebCore::Editor::insertAttachment):
(WebCore::Editor::handleAcceptedCandidate):
(WebCore::Editor::unifiedTextCheckerEnabled const):
(WebCore::Editor::toggleOverwriteModeEnabled):
(WebCore::Editor::styleForSelectionStart):
(WebCore::Editor::fontForSelection):
(WebCore::Editor::canCopyExcludingStandaloneImages const):
* Source/WebCore/editing/Editor.h:
(WebCore::Editor::protectedCompositionNode const):
(WebCore::Editor::document const):
(WebCore::Editor::protectedDocument const):
* Source/WebCore/editing/FormatBlockCommand.cpp:
(WebCore::FormatBlockCommand::FormatBlockCommand):
* Source/WebCore/editing/FormatBlockCommand.h:
(WebCore::FormatBlockCommand::create):
* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::IndentOutdentCommand):
(WebCore::IndentOutdentCommand::outdentParagraph):
* Source/WebCore/editing/IndentOutdentCommand.h:
(WebCore::IndentOutdentCommand::create):
* Source/WebCore/editing/InsertLineBreakCommand.cpp:
(WebCore::InsertLineBreakCommand::InsertLineBreakCommand):
(WebCore::InsertLineBreakCommand::doApply):
* Source/WebCore/editing/InsertLineBreakCommand.h:
(WebCore::InsertLineBreakCommand::create):
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::InsertListCommand::insertList):
(WebCore::InsertListCommand::InsertListCommand):
* Source/WebCore/editing/InsertListCommand.h:
* Source/WebCore/editing/InsertNestedListCommand.h:
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::InsertParagraphSeparatorCommand::InsertParagraphSeparatorCommand):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.h:
(WebCore::InsertParagraphSeparatorCommand::create):
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::InsertTextCommand):
* Source/WebCore/editing/InsertTextCommand.h:
(WebCore::InsertTextCommand::create):
(WebCore::InsertTextCommand::createWithMarkerSupplier):
* Source/WebCore/editing/ModifySelectionListLevel.cpp:
(WebCore::ModifySelectionListLevelCommand::ModifySelectionListLevelCommand):
(WebCore::IncreaseSelectionListLevelCommand::IncreaseSelectionListLevelCommand):
(WebCore::DecreaseSelectionListLevelCommand::DecreaseSelectionListLevelCommand):
* Source/WebCore/editing/ModifySelectionListLevel.h:
(WebCore::DecreaseSelectionListLevelCommand::create):
* Source/WebCore/editing/RemoveFormatCommand.cpp:
(WebCore::RemoveFormatCommand::RemoveFormatCommand):
* Source/WebCore/editing/RemoveFormatCommand.h:
(WebCore::RemoveFormatCommand::create):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplaceSelectionCommand::ReplaceSelectionCommand):
* Source/WebCore/editing/ReplaceSelectionCommand.h:
(WebCore::ReplaceSelectionCommand::create):
* Source/WebCore/editing/SimplifyMarkupCommand.cpp:
(WebCore::SimplifyMarkupCommand::SimplifyMarkupCommand):
* Source/WebCore/editing/SimplifyMarkupCommand.h:
(WebCore::SimplifyMarkupCommand::create):
* Source/WebCore/editing/SpellingCorrectionCommand.cpp:
(WebCore::SpellingCorrectionRecordUndoCommand::create):
(WebCore::SpellingCorrectionRecordUndoCommand::SpellingCorrectionRecordUndoCommand):
* Source/WebCore/editing/TextInsertionBaseCommand.cpp:
(WebCore::TextInsertionBaseCommand::TextInsertionBaseCommand):
* Source/WebCore/editing/TextInsertionBaseCommand.h:
(WebCore::TextInsertionBaseCommand::~TextInsertionBaseCommand):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::TextIterator::advance):
(WebCore::TextIterator::handleTextNode):
(WebCore::TextIterator::handleTextRun):
(WebCore::TextIterator::shouldRepresentNodeOffsetZero):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::TypingCommand):
(WebCore::TypingCommand::deleteSelection):
(WebCore::TypingCommand::deleteKeyPressed):
(WebCore::TypingCommand::forwardDeleteKeyPressed):
(WebCore::TypingCommand::insertText):
(WebCore::TypingCommand::insertLineBreak):
(WebCore::TypingCommand::insertParagraphSeparatorInQuotedContent):
(WebCore::TypingCommand::insertParagraphSeparator):
* Source/WebCore/editing/TypingCommand.h:
* Source/WebCore/editing/UnlinkCommand.cpp:
(WebCore::UnlinkCommand::UnlinkCommand):
* Source/WebCore/editing/UnlinkCommand.h:
(WebCore::UnlinkCommand::create):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionToAvoidCrossingEditingBoundaries):
* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::selectionInHTMLFormat):
(WebCore::Editor::writeSelectionToPasteboard):
(WebCore::Editor::writeSelection):
(WebCore::Editor::selectionInWebArchiveFormat):
(WebCore::Editor::replaceSelectionWithAttributedString):
(WebCore::Editor::webContentFromPasteboard):
(WebCore::Editor::takeFindStringFromSelection):
(WebCore::Editor::readSelectionFromPasteboard):
(WebCore::Editor::replaceNodeFromPasteboard):
* Source/WebCore/editing/ios/DictationCommandIOS.cpp:
(WebCore::DictationCommandIOS::DictationCommandIOS):
(WebCore::DictationCommandIOS::create):
* Source/WebCore/editing/ios/DictationCommandIOS.h:
* Source/WebCore/editing/ios/EditorIOS.mm:
(WebCore::Editor::setTextAsChildOfElement):
(WebCore::Editor::ensureLastEditCommandHasCurrentSelectionIfOpenForMoreTyping):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::platformCopyFont):
(WebCore::Editor::platformPasteFont):
(WebCore::Editor::selectionWillChange):

Canonical link: <a href="https://commits.webkit.org/268286@main">https://commits.webkit.org/268286@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e30b5bd4a03a3e71988fa28ad0e67e391bed460

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19236 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19652 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20250 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21126 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17998 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19454 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22923 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19778 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19457 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19524 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16725 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21995 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/16712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17511 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17772 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17686 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21824 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/18273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17406 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4597 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21763 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/18107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -498,29 +498,29 @@ void TextIterator::advance()
                     }
-                    next = nextSibling(m_behaviors, *m_currentNode);
+                    next = nextSibling(m_behaviors, *protectedCurrentNode());
                     if (next && isRendererVisible(m_currentNode->renderer(), m_behaviors))
```
Should this be `protectedCurrentNode()->renderer()`?

Perhaps you should capture the result of `protectedCurrentNode()` after the assignment to `m_currentNode` above so we don't have as much Ref churn?

brentfulgham commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -498,29 +498,29 @@ void TextIterator::advance()
                     bool haveRenderer = isRendererVisible(m_currentNode->renderer(), m_behaviors);
-                    RefPtr exitedNode = m_currentNode.get();
+                    auto exitedNode = protectedCurrentNode();
                     m_currentNode = parentNode;
```
Perhaps capturing the result of `protectedCurrentNode()` after this assignment would allow you to resume the same object in the rest of this method without having to create new Ref's at each call site?

brentfulgham commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -498,29 +498,29 @@ void TextIterator::advance()
-                bool pastEnd = nextNode(m_behaviors, *m_currentNode) == m_pastEndNode;
-                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *m_currentNode);
+                bool pastEnd = nextNode(m_behaviors, *protectedCurrentNode()) == m_pastEndNode;
+                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *protectedCurrentNode());
```
In lines 504 through 507 we construct Ref's for `protectedCurrentNode()` on each line. Maybe we could reduce cost by doing it once right before line 504? Naming might be tricky since you'd have to retrieve it again around like 514.

brentfulgham reviewed (approved):
A large patch! I think there is one place where we might benefit from making fewer Refs, but I'll defer to your judgement. r=me

cdumez committed:
Use more smart pointers in editing code
https://bugs.webkit.org/show_bug.cgi?id=261844

Reviewed by Brent Fulgham.

* Source/WebCore/editing/ApplyBlockElementCommand.cpp:
(WebCore::ApplyBlockElementCommand::ApplyBlockElementCommand):
* Source/WebCore/editing/ApplyBlockElementCommand.h:
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::ApplyStyleCommand):
* Source/WebCore/editing/ApplyStyleCommand.h:
(WebCore::ApplyStyleCommand::create):
* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::BreakBlockquoteCommand):
* Source/WebCore/editing/BreakBlockquoteCommand.h:
(WebCore::BreakBlockquoteCommand::create):
* Source/WebCore/editing/ChangeListTypeCommand.h:
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::CompositeEditCommand):
(WebCore::CompositeEditCommand::apply):
(WebCore::CompositeEditCommand::ensureComposition):
(WebCore::CompositeEditCommand::applyStyle):
(WebCore::CompositeEditCommand::insertParagraphSeparator):
(WebCore::CompositeEditCommand::insertLineBreak):
(WebCore::CompositeEditCommand::inputText):
(WebCore::CompositeEditCommand::deleteSelection):
(WebCore::CompositeEditCommand::deleteInsignificantText):
(WebCore::CompositeEditCommand::appendBlockPlaceholder):
(WebCore::CompositeEditCommand::insertBlockPlaceholder):
(WebCore::CompositeEditCommand::addBlockPlaceholderIfNeeded):
(WebCore::CompositeEditCommand::insertNewDefaultParagraphElementAt):
(WebCore::CompositeEditCommand::moveParagraphContentsToNewBlockIfNecessary):
(WebCore::CompositeEditCommand::moveParagraphWithClones):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/CompositeEditCommand.h:
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::CreateLinkCommand):
* Source/WebCore/editing/CreateLinkCommand.h:
(WebCore::CreateLinkCommand::create):
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::DeleteSelectionCommand::DeleteSelectionCommand):
(WebCore::DeleteSelectionCommand::removeNode):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
* Source/WebCore/editing/DeleteSelectionCommand.h:
(WebCore::DeleteSelectionCommand::create):
* Source/WebCore/editing/DictationCommand.cpp:
(WebCore::DictationCommand::DictationCommand):
(WebCore::DictationCommand::insertText):
(WebCore::DictationCommand::insertParagraphSeparator):
* Source/WebCore/editing/DictationCommand.h:
(WebCore::DictationCommand::create):
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::EditCommand::EditCommand):
(WebCore::SimpleEditCommand::SimpleEditCommand):
* Source/WebCore/editing/EditCommand.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::ClearTextCommand::ClearTextCommand):
(WebCore::ClearTextCommand::CreateAndApply):
(WebCore::TemporarySelectionChange::TemporarySelectionChange):
(WebCore::Editor::selectionForCommand):
(WebCore::Editor::behavior const):
(WebCore::Editor::client const):
(WebCore::Editor::canEdit const):
(WebCore::Editor::canEditRichly const):
(WebCore::Editor::canDHTMLCut):
(WebCore::Editor::canDHTMLCopy):
(WebCore::Editor::canCopy const):
(WebCore::Editor::canPaste const):
(WebCore::Editor::canDelete const):
(WebCore::Editor::shouldSmartDelete):
(WebCore::Editor::deleteWithDirection):
(WebCore::Editor::deleteSelectionWithSmartDelete):
(WebCore::Editor::clearText):
(WebCore::Editor::pasteAsPlainText):
(WebCore::Editor::pasteAsPlainTextBypassingDHTML):
(WebCore::Editor::replaceSelectionWithFragment):
(WebCore::Editor::selectedRange):
(WebCore::Editor::tryDHTMLCopy):
(WebCore::Editor::tryDHTMLCut):
(WebCore::Editor::shouldInsertText const):
(WebCore::Editor::hasBidiSelection const):
(WebCore::Editor::selectionUnorderedListState const):
(WebCore::Editor::selectionOrderedListState const):
(WebCore::Editor::insertOrderedList):
(WebCore::Editor::insertUnorderedList):
(WebCore::Editor::increaseSelectionListLevel):
(WebCore::Editor::increaseSelectionListLevelOrdered):
(WebCore::Editor::increaseSelectionListLevelUnordered):
(WebCore::Editor::decreaseSelectionListLevel):
(WebCore::Editor::removeFormattingAndStyle):
(WebCore::Editor::findEventTargetFromSelection const):
(WebCore::Editor::applyStyle):
(WebCore::Editor::applyParagraphStyle):
(WebCore::Editor::applyStyleToSelection):
(WebCore::Editor::applyParagraphStyleToSelection):
(WebCore::Editor::selectionStartHasStyle const):
(WebCore::Editor::selectionHasStyle const):
(WebCore::Editor::selectionStartCSSPropertyValue):
(WebCore::Editor::indent):
(WebCore::Editor::outdent):
(WebCore::Editor::appliedEditing):
(WebCore::Editor::unappliedEditing):
(WebCore::Editor::reappliedEditing):
(WebCore::Editor::clear):
(WebCore::Editor::insertText):
(WebCore::Editor::insertTextForConfirmedComposition):
(WebCore::Editor::insertTextWithoutSendingTextEvent):
(WebCore::Editor::insertLineBreak):
(WebCore::Editor::insertParagraphSeparator):
(WebCore::Editor::insertParagraphSeparatorInQuotedContent):
(WebCore::Editor::canCopyFont const):
(WebCore::Editor::performCutOrCopy):
(WebCore::Editor::paste):
(WebCore::Editor::pasteAsQuotation):
(WebCore::Editor::quoteFragmentForPasting):
(WebCore::Editor::changeSelectionListType):
(WebCore::Editor::simplifyMarkup):
(WebCore::Editor::copyURL):
(WebCore::Editor::copyImage):
(WebCore::Editor::revealSelectionIfNeededAfterLoadingImageForElement):
(WebCore::Editor::renderLayerDidScroll):
(WebCore::Editor::registerCustomUndoStep):
(WebCore::Editor::setBaseWritingDirection):
(WebCore::Editor::baseWritingDirectionForSelectionStart const):
(WebCore::Editor::selectComposition):
(WebCore::Editor::confirmOrCancelCompositionAndNotifyClient):
(WebCore::SetCompositionScope::SetCompositionScope):
(WebCore::Editor::setComposition):
(WebCore::Editor::ignoreSpelling):
(WebCore::Editor::learnSpelling):
(WebCore::Editor::advanceToNextMisspelling):
(WebCore::Editor::misspelledWordAtCaretOrRange const):
(WebCore::Editor::guessesForMisspelledWord const):
(WebCore::Editor::guessesForMisspelledOrUngrammatical):
(WebCore::Editor::markMisspellingsAfterTypingToWord):
(WebCore::Editor::isSpellCheckingEnabledInFocusedNode const):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
(WebCore::Editor::markAndReplaceFor):
(WebCore::Editor::updateMarkersForWordsAffectedByEditing):
(WebCore::Editor::rangeForPoint):
(WebCore::Editor::revealSelectionAfterEditingOperation):
(WebCore::Editor::setIgnoreSelectionChanges):
(WebCore::Editor::getCompositionSelection const):
(WebCore::Editor::transpose):
(WebCore::Editor::changeSelectionAfterCommand):
(WebCore::Editor::selectedText const):
(WebCore::Editor::selectedTextForDataTransfer const):
(WebCore::Editor::insertTextPlaceholder):
(WebCore::Editor::removeTextPlaceholder):
(WebCore::Editor::shouldChangeSelection const):
(WebCore::Editor::computeAndSetTypingStyle):
(WebCore::Editor::findString):
(WebCore::Editor::rangeOfString):
(WebCore::Editor::countMatchesForText):
(WebCore::Editor::respondToChangedSelection):
(WebCore::Editor::shouldDetectTelephoneNumbers const):
(WebCore::Editor::scanSelectionForTelephoneNumbers):
(WebCore::Editor::editorUIUpdateTimerFired):
(WebCore::Editor::selectionStartHasMarkerFor const):
(WebCore::Editor::stringForCandidateRequest const):
(WebCore::Editor::contextRangeForCandidateRequest const):
(WebCore::Editor::fontAttributesAtSelectionStart):
(WebCore::Editor::notifyClientOfAttachmentUpdates):
(WebCore::Editor::insertAttachment):
(WebCore::Editor::handleAcceptedCandidate):
(WebCore::Editor::unifiedTextCheckerEnabled const):
(WebCore::Editor::toggleOverwriteModeEnabled):
(WebCore::Editor::styleForSelectionStart):
(WebCore::Editor::fontForSelection):
(WebCore::Editor::canCopyExcludingStandaloneImages const):
* Source/WebCore/editing/Editor.h:
(WebCore::Editor::protectedCompositionNode const):
(WebCore::Editor::document const):
(WebCore::Editor::protectedDocument const):
* Source/WebCore/editing/FormatBlockCommand.cpp:
(WebCore::FormatBlockCommand::FormatBlockCommand):
* Source/WebCore/editing/FormatBlockCommand.h:
(WebCore::FormatBlockCommand::create):
* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::IndentOutdentCommand):
(WebCore::IndentOutdentCommand::outdentParagraph):
* Source/WebCore/editing/IndentOutdentCommand.h:
(WebCore::IndentOutdentCommand::create):
* Source/WebCore/editing/InsertLineBreakCommand.cpp:
(WebCore::InsertLineBreakCommand::InsertLineBreakCommand):
(WebCore::InsertLineBreakCommand::doApply):
* Source/WebCore/editing/InsertLineBreakCommand.h:
(WebCore::InsertLineBreakCommand::create):
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::InsertListCommand::insertList):
(WebCore::InsertListCommand::InsertListCommand):
* Source/WebCore/editing/InsertListCommand.h:
* Source/WebCore/editing/InsertNestedListCommand.h:
* Source/WebCore/editing/InsertParagraphSeparatorCommand.cpp:
(WebCore::InsertParagraphSeparatorCommand::InsertParagraphSeparatorCommand):
* Source/WebCore/editing/InsertParagraphSeparatorCommand.h:
(WebCore::InsertParagraphSeparatorCommand::create):
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::InsertTextCommand):
* Source/WebCore/editing/InsertTextCommand.h:
(WebCore::InsertTextCommand::create):
(WebCore::InsertTextCommand::createWithMarkerSupplier):
* Source/WebCore/editing/ModifySelectionListLevel.cpp:
(WebCore::ModifySelectionListLevelCommand::ModifySelectionListLevelCommand):
(WebCore::IncreaseSelectionListLevelCommand::IncreaseSelectionListLevelCommand):
(WebCore::DecreaseSelectionListLevelCommand::DecreaseSelectionListLevelCommand):
* Source/WebCore/editing/ModifySelectionListLevel.h:
(WebCore::DecreaseSelectionListLevelCommand::create):
* Source/WebCore/editing/RemoveFormatCommand.cpp:
(WebCore::RemoveFormatCommand::RemoveFormatCommand):
* Source/WebCore/editing/RemoveFormatCommand.h:
(WebCore::RemoveFormatCommand::create):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplaceSelectionCommand::ReplaceSelectionCommand):
* Source/WebCore/editing/ReplaceSelectionCommand.h:
(WebCore::ReplaceSelectionCommand::create):
* Source/WebCore/editing/SimplifyMarkupCommand.cpp:
(WebCore::SimplifyMarkupCommand::SimplifyMarkupCommand):
* Source/WebCore/editing/SimplifyMarkupCommand.h:
(WebCore::SimplifyMarkupCommand::create):
* Source/WebCore/editing/SpellingCorrectionCommand.cpp:
(WebCore::SpellingCorrectionRecordUndoCommand::create):
(WebCore::SpellingCorrectionRecordUndoCommand::SpellingCorrectionRecordUndoCommand):
* Source/WebCore/editing/TextInsertionBaseCommand.cpp:
(WebCore::TextInsertionBaseCommand::TextInsertionBaseCommand):
* Source/WebCore/editing/TextInsertionBaseCommand.h:
(WebCore::TextInsertionBaseCommand::~TextInsertionBaseCommand):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::TextIterator::advance):
(WebCore::TextIterator::handleTextNode):
(WebCore::TextIterator::handleTextRun):
(WebCore::TextIterator::shouldRepresentNodeOffsetZero):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::TypingCommand):
(WebCore::TypingCommand::deleteSelection):
(WebCore::TypingCommand::deleteKeyPressed):
(WebCore::TypingCommand::forwardDeleteKeyPressed):
(WebCore::TypingCommand::insertText):
(WebCore::TypingCommand::insertLineBreak):
(WebCore::TypingCommand::insertParagraphSeparatorInQuotedContent):
(WebCore::TypingCommand::insertParagraphSeparator):
* Source/WebCore/editing/TypingCommand.h:
* Source/WebCore/editing/UnlinkCommand.cpp:
(WebCore::UnlinkCommand::UnlinkCommand):
* Source/WebCore/editing/UnlinkCommand.h:
(WebCore::UnlinkCommand::create):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionToAvoidCrossingEditingBoundaries):
* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::selectionInHTMLFormat):
(WebCore::Editor::writeSelectionToPasteboard):
(WebCore::Editor::writeSelection):
(WebCore::Editor::selectionInWebArchiveFormat):
(WebCore::Editor::replaceSelectionWithAttributedString):
(WebCore::Editor::webContentFromPasteboard):
(WebCore::Editor::takeFindStringFromSelection):
(WebCore::Editor::readSelectionFromPasteboard):
(WebCore::Editor::replaceNodeFromPasteboard):
* Source/WebCore/editing/ios/DictationCommandIOS.cpp:
(WebCore::DictationCommandIOS::DictationCommandIOS):
(WebCore::DictationCommandIOS::create):
* Source/WebCore/editing/ios/DictationCommandIOS.h:
* Source/WebCore/editing/ios/EditorIOS.mm:
(WebCore::Editor::setTextAsChildOfElement):
(WebCore::Editor::ensureLastEditCommandHasCurrentSelectionIfOpenForMoreTyping):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::platformCopyFont):
(WebCore::Editor::platformPasteFont):
(WebCore::Editor::selectionWillChange):

Canonical link: https://commits.webkit.org/268286@main
